woocommerce/woocommerce-ios
# UL&S: Remove .startMagicLinkFlow segue

Ref. #2003 

This PR removes the segue `.startMagicLinkFlow` that sends a user to the "Send Magic Link?" screen. It now navigates the user programmatically.

Main PR changes here: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/256

## To test
1. Check out this branch
2. `rake dependencies`
3. Build and run
4. Log out if logged in
5. Log in with Jetpack > enter email address > next
6. **Expected:** the Magic Link screen will appear
7. Send Link > open the magic link* 
8. **Expected:** you are successfully navigated to the store list screen

**Step 7 Note:** If 2fa is active for a non-Automattic account, the magic link will ask for a 2fa code first. If the code is correct, then it will redirect you and reopen the WordPress app. If your login is an Automattic email account, you will be blocked from sending magic links.

### Do not merge until:
1. The Authenticator PR has been merged: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/256 ✔️
2. A new Authenticator release has been created ✔️  
3. The release is published on Cocoapods trunk ✔️
4. This PR points to the new release ✔️

Update release notes:
